File: week5/deserialization/python/send.py
# client.py
import json
import requests

# Data to send
user_data = {
    "name": "Ali",
    "age": 25,
    "is_admin": True
}

# Serialize to JSON
json_payload = json.dumps(user_data)

# Send to server
response = requests.post("http://localhost:5000/api/user", data=json_payload, headers={"Content-Type": "application/json"})

print("Server Response:", response.text)

File: week5/deserialization/nodejs/send.js
const express = require('express');
const app = express();
const port = 5000;

// Middleware to parse JSON
app.use(express.json());

app.post('/api/user', (req, res) => {
  const { name, age, is_admin } = req.body;

  console.log(`Received user: ${name}, Age: ${age}, Admin: ${is_admin}`);

  res.json({
    status: 'success',
    message: `Hello ${name}, your data was received!`
  });
});

app.listen(port, () => {
  console.log(`Server running at http://localhost:${port}`);
});
